From 4049f26b4ebf808766bd3301750cefa65193cf91 Mon Sep 17 00:00:00 2001
From: Sridharan S N <quic_sridsn@quicinc.com>
Date: Thu, 20 Jun 2024 17:00:01 +0530
Subject: [PATCH] hwtracing: coresight-tmc-etr: remove 1MB size check

In tmc-etr mode, reserved memory region size 1MB is used
as buffer size. But it is not common for all platforms
Hence remove 1MB size check.

Change-Id: I393aea1a49a724fbc64db6eaf4ca5721d1a6c3d1
Signed-off-by: Sridharan S N <quic_sridsn@quicinc.com>
---
 drivers/hwtracing/coresight/coresight-tmc-core.c | 3 ++-
 drivers/hwtracing/coresight/coresight-tmc-etr.c  | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/hwtracing/coresight/coresight-tmc-core.c b/drivers/hwtracing/coresight/coresight-tmc-core.c
index 211bb4c95b48..141e842e3c62 100644
--- a/drivers/hwtracing/coresight/coresight-tmc-core.c
+++ b/drivers/hwtracing/coresight/coresight-tmc-core.c
@@ -455,7 +455,8 @@ static void tmc_get_reserved_region(struct device *dev)
 	drvdata->etr_rsvd_vaddr = devm_ioremap(dev, rmem->base,rmem->size);
 	if (drvdata->etr_rsvd_vaddr) {
 		drvdata->etr_rsvd_paddr = rmem->base;
-		drvdata->rsvd_size =  rmem->size;
+		drvdata->rsvd_size = rmem->size;
+		drvdata->size = rmem->size;
 	}
 }
 
diff --git a/drivers/hwtracing/coresight/coresight-tmc-etr.c b/drivers/hwtracing/coresight/coresight-tmc-etr.c
index eb388b0db3e7..2c3fabb559be 100644
--- a/drivers/hwtracing/coresight/coresight-tmc-etr.c
+++ b/drivers/hwtracing/coresight/coresight-tmc-etr.c
@@ -927,7 +927,7 @@ static struct etr_buf *tmc_alloc_etr_buf(struct tmc_drvdata *drvdata,
 	 * Fallback to available mechanisms.
 	 *
 	 */
-	if (size == SZ_1M){
+	if (drvdata->etr_rsvd_paddr) {
 		rc =  tmc_etr_mode_alloc_buf(ETR_MODE_RESERVED, drvdata,
 					    etr_buf, node, pages);
 		goto err_check;
-- 
2.34.1

